lib/jetstream/api/consumer.ex

```diff
@@ -51,20 +234,48 @@ defmodule Jetstream.API.Consumer do
           total: non_neg_integer()
         }
 
-  @type t :: %__MODULE__{
-          stream_name: binary(),
-          name: binary()
-        }
+  @doc """
+  Creates a consumer. When consumer's `:durable_name` field is not set, the function
+  creates an ephemeral consumer. Otherwise, it creates a durable consumer.
+
+  ## Examples
+
+      iex> create(gnat, %Jetstream.API.Consumer{durable_name: "consumer", stream_name: "stream"})
+      {:ok, %{name: "consumer", stream_name: "stream", created: ~U[2022-02-23 14:21:42.876321Z]}}
```

Comment:
We may experiment with setting up doctests: https://elixir-lang.org/getting-started/mix-otp/docs-tests-and-with.html#doctests

I recommend creating separate test case modules for doctests, so for example: `Jetstream.API.ConsumerDocTest`.

Reply:
Hmm, doctests do explicitly compare the result which wouldn't be possible for us because the returned struct has the `created` timestamp. We could do something like this:

```
...
iex> with {:ok, %Jetstream.API.Consumer{}} <- result, do: :passed
:passed
```

but it wouldn't look good in the docs

Reply:
Hmm what about this?

```elixir
iex> {:ok, %{name: "consumer", stream_name: "stream", created: %DateTime{}} = create(gnat, %Jetstream.API.Consumer{durable_name: "consumer", stream_name: "stream"})
```

Output omitted deliberately.

Reply:
I didn't know the output can be omitted 😮